2048_fib.c

```diff
@@ -0,0 +1,588 @@
+		printf("%c  \t",(t++)+48);
+		while(c!=EOF && t<=20){
+			fseek(fp,-1,SEEK_CUR);
+			fgets(arr,100,fp);
+			printf("%s",arr);
+			c=fgetc(fp);
+			if(c!=EOF)
+				printf("%c  \t",(t++)+48);
+		}
+		fclose(fp);
+}
+
+void printGame(int *a,int n) {
+	printf("\n\n\t\t\t");
+	for (int i = 0; i < n; i++) {
+		for (int j = 0; j < n; j++) {
+			printf("%d\t", *(a+(i*n) + j));
+		}
+		printf("\n\t\t\t");
+	}
+	printf("\n");
+}
+
+void getName(char name[]) {
+		printf("enter nick-name: ");
+		scanf("%s",name);
+		int k = 0;
+		FILE *fp;
+			while (true) {
+				k = 0;
+				if(fp=fopen("fib_leaderboard.txt","r")){
+					if (nameExists(fp,name)) {
+						k = 1;
+					}
+					fclose(fp);
+				}
+				if (k == 0) {
+					if(fp=fopen("fib_state.txt","r")){
+						printf("I");
+						if (nameExists(fp,name)) {
+							k = 1;
+						}
+						fclose(fp);	
+					}
+				}
+				if (k == 0)
+					break;
+				printf("name is already taken choose other name: ");
+				scanf("%s",name);
+			}
+	}
+
+bool nameExists(FILE *fp,char name[]){
+	char c,t_name[20];
+	int k=0;
+	char temp[100]; // this is used woth fgets to move file pointer to next line
+	while(true){
+		k=0;
+		c=fgetc(fp);
+		while(c!=' '){
+			t_name[k++]=c;
+			c=fgetc(fp);
+		}
+		t_name[k]='\0';
+		if(!strcmp(t_name,name))
+			return 1;
+		fgets(temp,100,fp);
+		c=fgetc(fp);
+		if(c==EOF)
+			break;
+		else fseek(fp,-1,SEEK_CUR);
+	}
+	return 0;
+}
+
+void playGame(char name[],int *a,int moves,int n) {
+		while (true) {
+			moves++;
+			system("cls");
+			printGame(a,n);
+			if (won(a,n)) {
+				storeInFile(name,moves,n);
+				break;
+			}
+			if (furtherMovePossible(a,n)==false) {
+				printf("Game Over \n enter 1 - new game / 2 - leaderboard / 3 - exit\n: ");
+				int c ;
+				scanf("%d",&c);
+				if (c == 1) {
+					generateNewGame();
+					break;
+				}
+				if (c == 2) {
+					getLeaderBoard();
+					break;
+				}
+				break;
+			}
+			printf("enter 1 - UP / 2 - DOWN / 3 - RIGHT / 4 - LEFT\n\t\t\t 5 - save state of game\n: ");
+			int c ;
+			scanf("%d",&c);
+			if (c == 1){
+				moves++;
+				upMove(a,n);
+			}
+			else if (c == 2){
+				moves++;
+				downMove(a,n);
+			}
+			else if (c == 3){
+				moves++;
+				rightMove(a,n);
+			}
+			else if (c == 4){
+				moves++;
+				leftMove(a,n);
+			}
+			else if(c==5){
+				saveStateToFile(name,moves,n,a);
+				break;
+			}
+			else printf("enter correct option");
+			generateOneRandomly(a,n);
+		}
+	}
+
+void generateOneRandomly(int *a,int n) {
+		srand(time(0)); 
+		int i = rand()%n, j = rand()%n;
+		while ( *((a+i*n) + j) != 0) {
+			i = rand()%n;
+			j = rand()%n;
+		}
+		 *((a+i*n) + j) = 1;
+	}
+
+bool won(int *a,int n) {
+		int win = 2 * n * n;
+		for (int i = 0; i < n; i++) {
+			for (int j = 0; j < n; j++) {
+				if ( *((a+i*n) + j) == win) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+bool furtherMovePossible(int *a,int n) {
+		for (int i = 0; i < n; i++) {
+			for (int j = 0; j < n; j++) {
+				if (i == n - 1 && j == n - 1) {
```

Comment:
the same here can we extract if-else ladder to some other function, think of a suitable and expressive name.